tests/integration/helpers.py

```diff
@@ -464,8 +466,8 @@ async def start_server(ops_test: OpsTest, unit_name: str) -> None:
             with attempt:
                 if not await get_process_pid(ops_test, unit_name, "mysqld"):
                     raise Exception
-    except RetryError:
-        raise Exception("Failed to start server.")
+    except RetryError as e:
+        raise Exception("Failed to start server.") from e
```

Comment:
IMHO it's better to enable reraise in `Retrying` and remove the custom exception, since it makes it easier to figure out what went wrong.